jupyterhub/orm.py

```diff
@@ -676,9 +676,11 @@ def find(cls, db, token, *, kind=None):
     def new(
         cls,
         token=None,
+        *,
```

Comment:
Hmmm. I've forgot, what was this used to? To accept but ignore some arguments? Can you help me? It felt a bit complicated to google about.

Reply:
`*` ensures all following arguments are keyword _only_. In Python, by default all keyword args can be passed positionally, but when you have more than a couple of arguments that's hard to read. The `*` stops positional parsing, to ensure all these args are passed as keywords. The effect here: `token` is the only argument accepted positionally, all others must be passed `arg=value`. This was already the case, but it is now explicit. It's part of what allows inserting a new arg (scopes) somewhere other than the end. If it could be passed positionally, that would cause weird results.

It's a way to say "Don't call me with a bunch of positional arguments, it'll be confusing."

For example:

```python
def foo(a, b=5):
    return a * b

def bar(a, *, b=5):
    return a * b

foo(1) # OK, intended
foo(1, b=3) # OK, intended
foo(1, 3) # b passed positionally, not by name!
bar(1, 3) # fails with TypeError: bar() takes 1 positional argument but 2 were given
```

Reply:
[link for reference](https://peps.python.org/pep-3102/)

Reply:
Thank you @minrk!!!!! You made it very clear and comprehensible to me!